js/fieldmanager-loader.js

```diff
@@ -4,22 +4,55 @@
  * Ensures that metaboxes have loaded before initializing functionality.
  * @param {function} callback - The callback function to execute when the DOM is ready.
  */
-function fmLoadModule( callback ) {
+ function fmLoadModule( callback ) {
+	/**
+	 * Wraps the provided callback to add check for the Gutenberg editor. If this
+	 * is called within the Gutenberg editor, then the callback function will trigger
+	 * after the metaboxes are initialized.
+	 */
+	const wrappedCallback = () => {
+		if (document.querySelector('.block-editor-page')) {
+			const metaboxesInitializedUnsubscribe = wp.data.subscribe(() => {
+				if ( wp.data.select( 'core/edit-post' ).areMetaBoxesInitialized() ) {
+					/**
+					 * `areMetaBoxesInitialized` is called immediately before the
+					 * `MetaBoxesArea` component is rendered, which is where the metabox
+					 * HTML is moved from the hidden div and into the main form element.
+					 *
+					 * This means we need to delay the callback a bit to allow this
+					 * component to render such that the jQuery `:visible` selector
+					 * can work as expected.
+					 *
+					 * @link https://github.com/WordPress/gutenberg/blob/019d0a1b1883a5c3e5c9cdecc60bd5e546b60a1b/packages/edit-post/src/components/meta-boxes/index.js#L38-L45
+					 * @link https://github.com/WordPress/gutenberg/blob/d39949a3b9dc8e12d5f5d33b9091f14b93b37c8a/packages/edit-post/src/components/meta-boxes/meta-boxes-area/index.js#L34-L36
+					 */
+					setTimeout(() => {
+						callback();
+					}, 100);
```

Comment:
There must be a way to do this without the `setTimeout` call, let's keep at it until we identify what that is. `setTimeout` is too fragile for this.

Reply:
Totally agree :) I figured starting here is better than nothing for the time being.

Reply:
I just did some quick testing and something like this works. The only thing I do not like about this approach is that we are not checking any state within the subscribe function, but instead checking a DOM element. Ideally, the code we run within `wp.data.subscribe` would validate against the store, however, this might be a good option for now.

```js
	const wrappedCallback = () => {
		if (document.querySelector('.block-editor-page')) {
			const metaboxesInitializedUnsubscribe = wp.data.subscribe(() => {
				// Metaboxes are rendered and in the DOM.
				if ( document.querySelector('.edit-post-meta-boxes-area__container') ) {
					callback();

					// Clean up.
					metaboxesInitializedUnsubscribe();
				}
			});
		} else{
			callback();
		}
	};
```

Reply:
I think subscribing to the `areMetaBoxesInitialized` state is the right approach, but rather than using a `setTimeout`, I think we could wrap the callback in a polling function that checks for the existence of the initialized markup before firing our callback. The polling function would need to have an upper limit of the amount of times it would run, but should work great.